sse-component/src/main/java/org/wso2/identity/oidc/sse/api/stream/StreamController.java

```diff
@@ -0,0 +1,376 @@
+    @ApiOperation(value = "", notes = "Updates the current status of a stream")
+    public ResponseEntity<Stream> updateStatus(@RequestBody Stream status,
+                                               @RequestHeader(value = AUTHORIZATION) String accessToken) {
+
+        if (accessToken.isEmpty()) {
+            return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+        }
+        String token = accessToken.substring(7);
+        String id = getClientId(token);
+        Optional<Stream> stream = streamRepository.findById(id);
+
+        if (stream.isPresent()) {
+            Stream streamUpdate = stream.get();
+            streamUpdate.setStatus(status.getStatus());
+            streamRepository.save(streamUpdate);
+            return new ResponseEntity<>(status, HttpStatus.OK);
+        }
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Add subject to the stream.
+     *
+     * @param subject     Subject instance.
+     * @param accessToken Authorization token.
+     * @return Response entity with status code.
+     */
+    @PostMapping("subjects:add")
+    @ApiOperation(value = "", notes = "Retrieve current stream configuration")
+    public ResponseEntity addSubject(@RequestBody Subject subject,
+                                     @RequestHeader(value = AUTHORIZATION) String accessToken) {
+
+        if (accessToken.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+        String token = accessToken.substring(7);
+        String id = getClientId(token);
+        Optional<Stream> stream = streamRepository.findById(id);
+
+        if (stream.isPresent()) {
+            Stream streamUpdate = stream.get();
+            List<Subject> newSubjects = new ArrayList<>();
+
+            if (streamUpdate.getSubjects() != null) {
+                newSubjects.addAll(streamUpdate.getSubjects());
+            }
+            newSubjects.add(subject);
+            streamUpdate.setSubjects(newSubjects);
+            streamRepository.save(streamUpdate);
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    /**
+     * Remove subject to the stream.
+     *
+     * @param removeSubject Subject instance.
+     * @param accessToken   Authorization token.
+     * @return Response entity with status code.
+     */
+    @PostMapping("subjects:remove")
+    @ApiOperation(value = "", notes = "Add new subject to the stream")
+    public ResponseEntity<?> removeSubject(@RequestBody Subject removeSubject,
+                                           @RequestHeader(value = AUTHORIZATION) String accessToken) {
+
+        if (accessToken.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+        String token = accessToken.substring(7);
+        String id = getClientId(token);
+        Optional<Stream> stream = streamRepository.findById(id);
```

Comment:
shall we move the getting stream to a separate methd and reuse in all the methods? This logic also used in most of the methods